reana_workflow_controller/rest/workflows.py

```diff
@@ -1362,3 +1362,155 @@ def unshare_workflow(
     except Exception as e:
         logging.exception(str(e))
         return jsonify({"message": str(e)}), 500
+
+
+@blueprint.route("/workflows/<workflow_id_or_name>/share-status", methods=["GET"])
```

Comment:
Let's change `share-status` to `share_status`, to be consistent with the rest (e.g. we already have `move_files`, `retention_rules`, `disk_usage`, ...)